lib/services/storageManagement2/lib/operations/storageAccounts.js

```diff
@@ -52,10 +86,34 @@ function StorageAccounts(client) {
  *                      {object} [request]  - The HTTP Request object if an error did not occur.
  *
  *                      {stream} [response] - The HTTP Response stream if an error did not occur.
+ * 
+ *          {Promise}   A promise is returned.
+ *             @resolve {CheckNameAvailabilityResult} - The deserialized result object.
+ *             @reject {Error} - The error object.
  */
-StorageAccounts.prototype.checkNameAvailability = function (name, options, callback) {
+StorageAccounts.prototype.checkNameAvailability = function (name, options, optionalCallback) {
+  var client = this.client;
+  var self = this;
+  if (!optionalCallback && typeof options === 'function') {
+    optionalCallback = options;
+    options = null;
+  }
+  if (!optionalCallback) {
+    return new Promise((resolve, reject) => {
+      self._checkNameAvailability(name, options, (err, result, request, response) => {
+        if (err) return reject(err);
+        
+        return resolve(result);
+      });
+    });
+  } else {
+    return self._checkNameAvailability(name, options, optionalCallback);
+  }
+};
+
+function _checkNameAvailability(name, options, callback) {
   var client = this.client;
```

Comment:
I have just prorotyped for checkNameAvailability() method. All the changes from here on in this file are a result of code formatting in vscode. Please ignore them.

Do take a look at the test.js file below.